clang/include/clang/AST/Type.h

```diff
@@ -480,6 +480,10 @@ class Qualifiers {
            // Otherwise in OpenCLC v2.0 s6.5.5: every address space except
            // for __constant can be used as __generic.
            (A == LangAS::opencl_generic && B != LangAS::opencl_constant) ||
+           // For USM extension we define global_device and global_host address
+           // spaces, which are a subset of __global.
```

Comment:
These address spaces are not only for USM, right?

Reply:
Well, yeah. It's kinda hard here to make a good reference while spec isn't yet published :)

Reply:
At least you can link to the description of the various public USM extensions :-)